Code/autopkglib/MunkiImporter.py

```diff
@@ -539,23 +577,44 @@ def main(self):
             return
 
         # copy pkg/dmg to repo
-        relative_path = self.copy_item_to_repo(pkginfo)
+        uploaded_pkgpath = munkiimportlib.copy_item_to_repo(
+            repo,
+            self.env["pkg_path"],
+            pkginfo.get("version"),
+            self.env.get("repo_subdirectory", ""),
+        )
         # adjust the installer_item_location to match the actual location
         # and name
-        pkginfo["installer_item_location"] = relative_path
+        pkginfo["installer_item_location"] = uploaded_pkgpath.partition("/")[2]
 
         if self.env.get("uninstaller_pkg_path"):
-            relative_uninstall_path = self.copy_item_to_repo(
-                pkginfo, uninstaller_pkg=True
+            uploaded_uninstall_path = munkiimportlib.copy_item_to_repo(
+                repo,
+                self.env["uninstaller_pkg_path"],
+                pkginfo.get("version"),
+                self.env.get("repo_subdirectory", ""),
             )
-            pkginfo["uninstaller_item_location"] = relative_uninstall_path
+            pkginfo["uninstaller_item_location"] = uploaded_uninstall_path.partition(
+                "/"
+            )[2]
             pkginfo["uninstallable"] = True
 
+        # extract and import icon if requested
+        self.env["icon_repo_path"] = ""
+        if self.env.get("extract_icon"):
+            if not munkiimportlib.icon_exists_in_repo(repo, pkginfo):
+                imported_icon_path = munkiimportlib.extract_and_copy_icon(
+                    repo, self.env["pkg_path"], pkginfo, import_multiple=False
+                )
+                if imported_icon_path:
+                    self.output("Copied icon to %s" % imported_icon_path)
+                    self.env["icon_repo_path"] = imported_icon_path
+
```

Comment:
This is brand new functionality that didn't previously exist in MunkiImporter. You'll also have to update the description and potentially parts of the AutoPkg wiki to describe this updated functionality, and how to use it.

Use f-strings instead of %s string interpolation here.